libs/api/metadata-converter/src/lib/iso19115-3/read-parts.ts

```diff
@@ -1,6 +1,197 @@
+  return findNestedElement('mdb:distributionInfo', 'mrd:MD_Distribution')
+}
+
+// from cit:CI_Responsibility
+export function extractOrganization(): ChainableFunction<
+  XmlElement,
+  Organization
+> {
+  const getOrganization = findNestedElement('cit:party', 'cit:CI_Organisation')
+  const getUrl = pipe(
+    getOrganization,
+    findNestedElements(
+      'cit:contactInfo',
+      'cit:CI_Contact',
+      'cit:onlineResource',
+      'cit:CI_OnlineResource',
+      'cit:linkage'
+    ),
+    getAtIndex(0),
+    extractUrl()
+  )
+  return pipe(
+    combine(
+      pipe(
+        getOrganization,
+        findChildElement('cit:name', false),
+        extractCharacterString()
+      ),
+      getUrl
+    ),
+    map(([name, website]) => ({
+      name,
+      ...(website && { website }),
+    }))
+  )
+}
+
+// from cit:CI_Responsibility
+export function extractIndividuals(): ChainableFunction<
+  XmlElement,
+  Array<Individual>
+> {
+  const getRole = pipe(findChildElement('cit:role'), extractRole())
+  const getIndividual = pipe(
+    findChildElement('cit:individual'),
+    findChildElement('cit:CI_Individual')
+  )
+  const getPosition = pipe(
+    getIndividual,
+    findChildElement('cit:positionName'),
+    extractCharacterString()
+  )
+  const getNameParts = pipe(
+    getIndividual,
+    findChildElement('cit:name'),
+    extractCharacterString(),
+    map((fullName) => {
+      if (!fullName) return []
+      const parts = fullName.split(/\s+/)
+      if (!parts.length) return [fullName, null]
+      const first = parts.shift()
+      return [first, parts.join(' ')]
+    })
+  )
+  const getOrganization = extractOrganization()
+  const getContacts = pipe(
+    findNestedElement('cit:party', 'cit:CI_Organisation'),
+    findChildrenElement('cit:contactInfo'),
+    mapArray(findChildElement('cit:CI_Contact'))
+  )
+  const getAddressRoots = pipe(
+    getContacts,
+    mapArray(findNestedElement('cit:address', 'cit:CI_Address'))
+  )
+  const getAddresses = pipe(
+    getAddressRoots,
+    mapArray(
+      combine(
+        pipe(
+          findChildElement('cit:deliveryPoint', false),
+          extractCharacterString()
+        ),
+        pipe(findChildElement('cit:city', false), extractCharacterString()),
+        pipe(
+          findChildElement('cit:postalCode', false),
+          extractCharacterString()
+        ),
+        pipe(findChildElement('cit:country', false), extractCharacterString())
+      )
+    ),
+    mapArray((parts) => parts.filter((p) => !!p).join(', '))
+  )
+  const getPhones = pipe(
+    getContacts,
+    mapArray(findNestedElement('cit:phone', 'cit:CI_Telephone', 'cit:number')),
+    mapArray(extractCharacterString())
+  )
+  const getEmails = pipe(
+    getAddressRoots,
+    mapArray(
+      pipe(
+        findChildElement('cit:electronicMailAddress', false),
+        extractCharacterString(),
+        map((email) => (email === null ? 'missing@missing.com' : email))
```

Comment:
Wouldn't testing for truthy email be more prudent? What about spaces?

Reply:
that would be up to the quality of the metadata; this is a text field so it can contain anything, even an empty string; in that case, I think it's better to keep the email as an empty string and have it show up like so, because that's what the metadata really contains